File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/LeaveApproval.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using HRManagementSystem.Domain.Enums;

namespace HRManagementSystem.Domain.Entities;
public class LeaveApproval
{
    public int Id { get; set; }

    public int LeaveRequestId { get; set; }
    public int ApproverId { get; set; }   // ✅ int instead of string

    public LevelApproval Level { get; set; }
    public LeaveStatus Status { get; set; } = LeaveStatus.Pending;
    public DateTime? ActionDate { get; set; }

    // 🔗 Navigation
    public LeaveRequest LeaveRequest { get; set; }
    public Employee Approver { get; set; }
}

File: frontend/src/app/models/leaveApproval.ts
export interface LeaveApproval {
  id: number;
  leaveRequestId: number;
  approverId: number;
  approverName: string;
  status: number; // 0: Pending, 1: Approved, 2: Rejected
  comments?: string;
  actionDate?: string;
  stepLevel: number;
  isFinalStep: boolean;
}

export interface LeaveApprovalAction {
  leaveRequestId: number;
  approverId: number;
}
